cron/notification-for-bekrachtigde-mandataris.ts

```diff
@@ -2,72 +2,149 @@ import { CronJob } from 'cron';
+      console.log(
+        'DEBUG: Starting cronjob to send notifications for active mandatarissen without besluit.',
+      );
+      if (running) {
+        return;
+      }
+      await handleMandatarissen();
+    },
+  });
+}
 
-async function handleEffectieveMandatarissen() {
-  const mandatarissen =
-    await fetchEffectiveMandatarissenWithoutBesluitAndNotification();
+export const cronjob = job;
+
+async function handleBatchOfMandatarissen(keyOfRun: string) {
+  const mandatarissen = await fetchActiveMandatarissenWithoutBesluit();
 
   if (mandatarissen.length == 0) {
-    return;
+    return mandatarissen;
   }
 
   await createBulkNotificationMandatarissenWithoutBesluit(
-    SUBJECT,
+    SUBJECT_DECISION,
     mandatarissen,
+    keyOfRun,
   );
 
-  if (SEND_EMAILS) {
-    const grouped_mandatarissen = mandatarissen.reduce((acc, mandataris) => {
-      const { graph } = mandataris;
-      if (!acc[graph]) {
-        acc[graph] = [];
-      }
-      acc[graph].push(mandataris);
-      return acc;
-    }, {});
-    for (const key in grouped_mandatarissen) {
-      const mandataris_group = grouped_mandatarissen[key];
-      const email = await getContactEmailForMandataris(
-        mandataris_group.at(0)?.uri,
-      );
-      if (email) {
-        await sendMissingBekrachtigingsmail(email, mandataris_group);
-      }
-    }
+  return mandatarissen;
+}
+
+export async function handleMandatarissen() {
+  running = true;
+  const keyOfRun = uuid();
+  let moreToDo = true;
+  while (moreToDo) {
+    const processed = await handleBatchOfMandatarissen(keyOfRun);
+    moreToDo = processed.length > 0;
   }
+  await sendEmailsForNotifications(keyOfRun);
   running = false;
 }
 
+async function sendEmailsForNotifications(keyOfRun: string) {
+  if (!SEND_EMAILS) {
+    return;
+  }
+  const graphs = await getGraphsWithMandatarissenRequiringMail(keyOfRun);
+  for (const graph of graphs) {
+    const mandatarissen = await getMandatarissenToNotifyAbout(graph, keyOfRun);
+    const email = await getContactEmailForMandataris(mandatarissen[0]?.uri);
+    if (email) {
+      await sendMissingBekrachtigingsmail(email, mandatarissen);
+    }
+  }
+}
+
+async function getGraphsWithMandatarissenRequiringMail(keyOfRun: string) {
+  const query = `
+    PREFIX ext: <http://mu.semte.ch/vocabularies/ext/>
+    PREFIX org: <http://www.w3.org/ns/org#>
+    PREFIX mandaat: <http://data.vlaanderen.be/ns/mandaat#>
+    PREFIX dct: <http://purl.org/dc/terms/>
+
+    SELECT DISTINCT ?g WHERE {
+      GRAPH ?g {
+        ?mandataris a mandaat:Mandataris .
+        ?notification a ext:SystemNotification ;
+          dct:subject ${sparqlEscapeString(SUBJECT_DECISION)} ;
+          ext:generatedByRun ${sparqlEscapeString(keyOfRun)} ;
+          ext:notificationLink / ext:linkedTo ?mandataris .
+      }
+      ?g ext:ownedBy ?eenheid.
+    }
+  `;
+  const result = await querySudo(query);
+  return getSparqlResults(result).map((term) => term.g.value);
```

Comment:
Are we sure we always find a mandataris with a notification in the graph?
```suggestion
  return getSparqlResults(result).map((term) => term.g?.value);

```

Reply:
that's how sparql works: if you get a result (and the var isn't part of an optional) it's guaranteed to be there